.github/workflows/unit-tests.yml

```diff
@@ -41,35 +41,30 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 11.8.0
     - uses: actions/setup-java@v1
       with:
         java-version: '8' # The JDK version needed by hail
         java-package: jdk # (jre, jdk, or jdk+fx) - defaults to jdk
         architecture: x64 # (x64 or x86) - defaults to x64
+    - name: Install JS Dependencies
+      run: npm install || true
+      working-directory: ./ui
+    - name: Javascript/NPM Tests
+      run: npm test
+      working-directory: ./ui
+    - name: build production app for python tests
+      run: CI= npm run build
```

Comment:
In our last build, that took 1m29s, and the entire end to end pipeline took 6m52s -- I ended up combining the pipelines here since it felt it might save time overall to only run the `npm install` once. It might be worth looking at the python tests that require this build, and seeing if we can refactor those like you mentioned in standup. I think it would be a plus overall to be able to run the python tests without needing the JS stack.

Another option is I can maybe add a prerequisite step that runs the npm stuff first, and then re-uses the resulting node_modules and build output in parallel for the node and python tests afterward. Though, I don't think will actually save that much time.

Reply:
I think 6+ minutes to get test results back every time we put up a PR is a bit slower than I really want. I know we don't want to block this PR indefinitely, but I think what I would prefer is if you could take out the tests that rely on the assets for now, and take out this extra step here, and then I can work in parallel/ shortly after this goes live on getting those tests re-enabled (or determine if we really need them at all)

Reply:
Actually, since we are leaving the assets in on this PR, then we can probably just take this github actions change out for now. Does it make more sense to do that, and then update the test pipeline in the PR where we deprecate assets and/or refactor the tests?

Reply:
yes that makes sense to me